index.js

```diff
@@ -896,7 +898,9 @@ function buildValue (context, location, input) {
     }
 
     code += `
-      else throw new TypeError(\`The value of '${schemaRef}' does not match schema definition.\`)
+      else throw Object.assign(new TypeError(\`The value of '${schemaRef}' does not match schema definition.\`), {
+        validationErrors: errors
+      })
```

Comment:
Why not use `cause` and avoid the Object.assign?

```suggestion
      else throw new TypeError(\`The value of '${schemaRef}' does not match schema definition.\`, {
        cause: errors
      })
```